kernel/src/main/java/org/cristalise/kernel/entity/ItemImplementation.java

```
@@ -382,7 +382,16 @@ public String queryLifeCycle(SystemKey agentId, boolean filter)
 
             if (secMan.isShiroEnabled()) {
                 for (Job j: jobs) {
-                    if (secMan.checkPermissions(agent, (Activity) wf.search(j.getStepPath()), mItemPath)) jobBag.list.add(j);
+                    Activity act =  (Activity) wf.search(j.getStepPath());
+                    if (secMan.checkPermissions(agent, act, mItemPath)) {
+                        try {
+                            //Throws AccessRightsException if Job requires specific Role that agent does not have
+                            j.getTransition().getPerformingRole(act, agent);
+                            jobBag.list.add(j);
+                        }
+                        catch (AccessRightsException e) {

```

add a comment in the `catch` block to explain why the exception can be ignored